src/common.js

```diff
@@ -151,13 +151,13 @@ export function resolveUrl (relUrl, parentUrl) {
   return resolveIfNotPlainOrUrl(relUrl, parentUrl) || (relUrl.indexOf(':') !== -1 ? relUrl : resolveIfNotPlainOrUrl('./' + relUrl, parentUrl));
 }
 
-function resolveAndComposePackages (packages, outPackages, baseUrl, parentMap) {
+function resolveAndComposePackages (packages, outPackages, baseUrl, parentMap, parentUrl) {
   for (let p in packages) {
-    const resolvedLhs = resolveIfNotPlainOrUrl(p, baseUrl) || p;
+    const resolvedLhs = resolveIfNotPlainOrUrl(p, parentUrl != undefined ? parentUrl : baseUrl) || p;
```

Comment:
If we are happy with `!=` semantics then `scopeUrl || baseUrl` is also fine :)

Reply:
`scopeUrl || baseUrl` wouldn't will return `baseURL` id `scopeURL` is an empty string, I'm not sure if an empty string can occur in that position though

Reply:
Good point. Yes best not to fret over these details so happy to leave as-is. An empty string is permitted, but the assumption in this code here is that the URLs are already normalized so should never be the empty string I believe.